### DISCUSSION TITLE
What is the purpose of the `rearchive` option?

### DISCUSSION THREAD

**Discussion by andreaskienast** [composer/satis#723] [Q&A]
Hello folks.

We currently tweak our Satis setup, and on [this page](https://composer.github.io/satis/config) I found the option `rearchive` that defaults to `true`. Unfortunately, this option is not explained in the [documentation](https://getcomposer.org/doc/articles/handling-private-packages.md) linked in the [README](https://github.com/composer/satis/blob/023a9a47912f9e3af0bba8671e864ce695f35c64/README.md).

I'd like to know if this is a private API we should not rely on (perhaps for Private Packagist) and if there are any adverse side effects when disabling this option.

I discovered two things: the download is obviously faster as re-archiving is skipped, but at the trade of larger archives, at least in my case. AFAIK enabling `rearchive` leads to different checksums when the package needs to get downloaded and thus re-archived again.

**Comment by alcohol** [comment]
Honestly, I'm not really sure anymore.. the documentation has been lacking for a long time and is not kept up to date at all. The github pages hosted one was an attempt I made a long time ago to remedy that, but alas..
